ProjectStructure.md

```diff
@@ -2,8 +2,147 @@
 
 This chapter outlines the directory structures and various other basic requirements for iOS & Android projects.
 
+
+
 ## iOS
 
+Project structure is the first thing we should pay attention when starting a project. Generally, there are a few rules which every iOS team member should follow.
+
+
+
+- **GIT** 
+
+  All projects in Hipo will be hosted in 'Github'. So, first thing we should do is to put a standard *.gitignore* file into the project. We do not push installed pods (CocoaPods) to git, therefore; do NOT forget to add the restriction into gitignore because, before, we had experienced that some pods might have too much size to be hosted in Github. Also, it would be nice to add a *README* file with general info about the project and how to setup development environment.
+
+      Sample **.gitignore** 
+
+```
+# OS X temporary files that should never be committed
+.DS_Store
+.Trashes
+*.swp
+
+# Xcode
+*.pbxuser
+!default.pbxuser
+*.mode1v3
+!default.mode1v3
+*.mode2v3
+!default.mode2v3
+*.perspectivev3
+!default.perspectivev3
+*.xcworkspace
+!default.xcworkspace
+xcuserdata
+*.moved-aside
+compile_commands.json
+
+# CocoaPods
+build/SampleProject/Pods/
+
+## Build generated
+build/SampleProject/DerivedData
+```
+
+
+- **COCOAPODS**
+
+  It is possible for a project not to have *CocoaPods* integrated. However, almost every project needs to benefit open-source libraries somewhere in the development process. For that, it is highly recommended to start a project with *CocoaPods* integrated.
+
+  If there is an open-source library, you want to use in the project, you must add it as a pod unless there is a specific reason not to do. It would be nice to write it down to *README*. Also, if you want to add a pod, but it  does not meet all your requirements and you need to change the code, first you fork the project into Hipo repo, make your changes, then add ours into *podfile*. 
+
+​      Sample **Podfile** 
+
+```
+platform :ios, '9.0'
+
+use_frameworks!
+
+def main_pods
```

Comment:
Let's change the pods there with the ones in Moment because it has some useful parameters and it is nice to group the pods in a logical order.

  #Analytics
  pod 'Intercom'
  pod 'Mixpanel-swift'
    
  #Debug
  pod 'Reveal-SDK', :configurations => ['Debug']

  #Layout
  pod 'PureLayout'
    
  #Model
  pod 'Mantle'
  pod 'KVOController'
    
  #Networking
  pod 'AWSS3'
  pod 'HIPNetworking', :git =>  'https://github.com/Hipo/HIPNetworking.git'
  pod 'Kingfisher', '~> 4.0'
    
  #View
  pod 'ActiveLabel', :git => 'https://github.com/Hipo/ActiveLabel.swift.git', :branch => 'swift-4-migration'
  pod 'CCHLinkTextView'
  pod 'HIPImageCropper'
  pod 'VENTokenField', :git =>   'https://github.com/Hipo/VENTokenField.git'
  pod 'PullToRefresher', :git => 'https://github.com/Hipo/PullToRefresh.git'
  pod 'SZTextView'
    
  #Utilities
  pod 'DeviceUtil'
  pod 'SSKeychain'
  pod 'PhoneNumberKit', '~> 2.1'
  pod 'AnyFormatKit', :git => 'https://github.com/Hipo/AnyFormatKit'
    
  #Structure
  pod 'IGListKit', '~> 3.0'